import{_ as c,c as p,J as a,m as t,a as e,w as u,V as i,E as s,o as h}from"./chunks/framework.WH0rnJL5.js";const d="/assets/truenas-scale-bluefin-encounters-errors-when-updating-apps-screenshot-1.7DlK1STk.png",m="/assets/truenas-scale-bluefin-encounters-errors-when-updating-apps-screenshot-2.kJTjXCAu.png",g="/assets/truenas-scale-bluefin-encounters-errors-when-updating-apps-screenshot-3.4Pzn4x4w.png",E=JSON.parse('{"title":"TrueNAS SCALE 发行版本 Bluefin 在更新 TrueCharts 应用时遭遇错误","description":"","frontmatter":{"tags":["基础设施/Homelab","基础设施/Homelab/NAS","软件/TrueNAS-SCALE","软件/TrueNAS","运维/云原生/Kubernetes","运维/云原生/Kubernetes/K3s","计算机/信息技术/安全","命令行/kubectl","软件/云原生/kubectl"]},"headers":[],"relativePath":"笔记/🧱 基础设施/💾 TrueNAS/TrueNAS SCALE 发行版本 Bluefin 在更新 TrueCharts 应用时遭遇错误.md","filePath":"笔记/🧱 基础设施/💾 TrueNAS/TrueNAS SCALE 发行版本 Bluefin 在更新 TrueCharts 应用时遭遇错误.md"}'),_={name:"笔记/🧱 基础设施/💾 TrueNAS/TrueNAS SCALE 发行版本 Bluefin 在更新 TrueCharts 应用时遭遇错误.md"},b=t("h1",{id:"truenas-scale-发行版本-bluefin-在更新-truecharts-应用时遭遇错误",tabindex:"-1"},[e("TrueNAS SCALE 发行版本 Bluefin 在更新 TrueCharts 应用时遭遇错误 "),t("a",{class:"header-anchor",href:"#truenas-scale-发行版本-bluefin-在更新-truecharts-应用时遭遇错误","aria-label":'Permalink to "TrueNAS SCALE 发行版本 Bluefin 在更新 TrueCharts 应用时遭遇错误"'},"​")],-1),k=i('<h2 id="第一次更新之后遇到的错误" tabindex="-1">第一次更新之后遇到的错误 <a class="header-anchor" href="#第一次更新之后遇到的错误" aria-label="Permalink to &quot;第一次更新之后遇到的错误&quot;">​</a></h2><h3 id="错误详情" tabindex="-1">错误详情 <a class="header-anchor" href="#错误详情" aria-label="Permalink to &quot;错误详情&quot;">​</a></h3><p>升级 TrueNAS SCALE 的应用（K3s Chart App）的时候遇到了这样的错误：</p><div class="language-txt vp-adaptive-theme"><button title="Copy Code" class="copy"></button><span class="lang">txt</span><pre class="shiki shiki-themes github-light one-dark-pro vp-code"><code><span class="line"><span>CallError: [EFAULT] Failed to upgrade chart release: Error: UPGRADE FAILED: template: scrutiny/templates/common.yaml:1:3: executing &quot;scrutiny/templates/common.yaml&quot; at &lt;include &quot;tc.v1.common.loader.all&quot; .&gt;: error calling include: template: scrutiny/charts/common/templates/loader/_all.tpl:6:6: executing &quot;tc.v1.common.loader.all&quot; at &lt;include &quot;tc.v1.common.loader.apply&quot; .&gt;: error calling include: template: scrutiny/charts/common/templates/loader/_apply.tpl:47:6: executing &quot;tc.v1.common.loader.apply&quot; at &lt;include &quot;tc.v1.common.spawner.pvc&quot; .&gt;: error calling include: template: scrutiny/charts/common/templates/spawner/_pvc.tpl:25:10: executing &quot;tc.v1.common.spawner.pvc&quot; at &lt;include &quot;tc.v1.common.lib.persistence.validation&quot; (dict &quot;rootCtx&quot; $ &quot;objectData&quot; $objectData)&gt;: error calling include: template: scrutiny/charts/common/templates/lib/storage/_validation.tpl:18:43: executing &quot;tc.v1.common.lib.persistence.validation&quot; at &lt;$objectData.static.mode&gt;: nil pointer evaluating interface {}.mode</span></span></code></pre></div><p>看着密密麻麻，但是实际上最关键的错误是这个：</p><div class="language-txt vp-adaptive-theme"><button title="Copy Code" class="copy"></button><span class="lang">txt</span><pre class="shiki shiki-themes github-light one-dark-pro vp-code"><code><span class="line"><span>executing &quot;tc.v1.common.lib.persistence.validation&quot; at &lt;$objectData.static.mode&gt;: nil pointer evaluating interface {}.mode</span></span></code></pre></div>',6),C=i('<h3 id="修复-nil-pointer-evaluating-interface-mode-错误" tabindex="-1">修复 <code>nil pointer evaluating interface {}.mode</code> 错误 <a class="header-anchor" href="#修复-nil-pointer-evaluating-interface-mode-错误" aria-label="Permalink to &quot;修复 `nil pointer evaluating interface {}.mode` 错误&quot;">​</a></h3><p>导航到 TrueNAS 的「系统设置」，然后点选「更新」：</p><p><img src="'+d+'" alt=""></p><p>在更新频道下拉框中选择现在最新的 TrueNAS SCALE 发行版「Cobia」将 Bluefin 更新到 Cobia：</p><p><img src="'+m+'" alt=""></p><p>更新完成之后再次前往左侧的「应用」选单即可：</p><p><img src="'+g+'" alt=""></p><h2 id="更新为-cobia-之后的-k3s-应用错误" tabindex="-1">更新为 Cobia 之后的 K3s 应用错误 <a class="header-anchor" href="#更新为-cobia-之后的-k3s-应用错误" aria-label="Permalink to &quot;更新为 Cobia 之后的 K3s 应用错误&quot;">​</a></h2><p>更新之后的 TrueNAS SCALE 大变样了，看他们改造了很多应用（K3s）相关的界面和使用体验。</p><p>更新之后再次检查「应用」选单可以发现应用会出现卡在「Deploying（正在部署）」的状态里面，如果看右边的 Kubernetes 事件列表或者直接通过命令行 <code>describe deployment</code>，会发现提示说：</p><div class="language-shell vp-adaptive-theme"><button title="Copy Code" class="copy"></button><span class="lang">shell</span><pre class="shiki shiki-themes github-light one-dark-pro vp-code"><code><span class="line"><span style="--shiki-light:#6F42C1;--shiki-dark:#61AFEF;">MountVolume.SetUp</span><span style="--shiki-light:#032F62;--shiki-dark:#98C379;"> failed for volume &quot;pvc-9062b24f-39e7-4809-bb4e-58c73bfdf052&quot;</span><span style="--shiki-light:#032F62;--shiki-dark:#98C379;"> : kubernetes.io/csi: mounter.SetUpAt failed to get CSI client: driver name zfs.csi.openebs.io not found in the list of registered CSI drivers</span></span></code></pre></div><div class="language-shell vp-adaptive-theme"><button title="Copy Code" class="copy"></button><span class="lang">shell</span><pre class="shiki shiki-themes github-light one-dark-pro vp-code"><code><span class="line"><span style="--shiki-light:#6F42C1;--shiki-dark:#61AFEF;">Warning</span><span style="--shiki-light:#032F62;--shiki-dark:#98C379;">  FailedMount  </span><span style="--shiki-light:#005CC5;--shiki-dark:#D19A66;">77</span><span style="--shiki-light:#032F62;--shiki-dark:#98C379;">s </span><span style="--shiki-light:#24292E;--shiki-dark:#ABB2BF;">(</span><span style="--shiki-light:#6F42C1;--shiki-dark:#61AFEF;">x25</span><span style="--shiki-light:#032F62;--shiki-dark:#98C379;"> over </span><span style="--shiki-light:#005CC5;--shiki-dark:#D19A66;">80</span><span style="--shiki-light:#032F62;--shiki-dark:#98C379;">s</span><span style="--shiki-light:#24292E;--shiki-dark:#ABB2BF;">)  </span><span style="--shiki-light:#032F62;--shiki-dark:#98C379;">kubelet            MountVolume.SetUp failed for volume &quot;pvc-9062b24f-39e7-4809-bb4e-58c73bfdf052&quot;</span><span style="--shiki-light:#032F62;--shiki-dark:#98C379;"> : kubernetes.io/csi: mounter.SetUpAt failed to get CSI client: driver name zfs.csi.openebs.io not found in the list of registered CSI drivers</span></span></code></pre></div><p>这个提示是说应用（K3s）引用的 PVC（持久卷申领）对应的 CSI（容器存储接口） <code>zfs.csi.openebs.io</code> 不存在了，虽然我还没有具体调查原因，但是推测是因为 CSI 的名字更新了，所以导致了这样的问题，可以通过更新 TrueCharts 来解决这个问题。</p><p>在应用列表中选择更新应用，然后选择最新版本的 TrueCharts 点选更新，更新之后应该就没有问题了。</p><h2 id="贡献者" tabindex="-1">贡献者 <a class="header-anchor" href="#贡献者" aria-label="Permalink to &quot;贡献者&quot;">​</a></h2>',15),f=t("h2",{id:"文件历史",tabindex:"-1"},[e("文件历史 "),t("a",{class:"header-anchor",href:"#文件历史","aria-label":'Permalink to "文件历史"'},"​")],-1);function v(A,S,T,y,q,N){const n=s("NolebasePageProperties"),o=s("VPNolebaseInlineLinkPreview"),r=s("NolebaseGitContributors"),l=s("NolebaseGitChangelog");return h(),p("div",null,[b,a(n),k,t("p",null,[e("查了一下资料之后发现 TrueNAS 社区有人提到了： "),a(o,{href:"https://www.truenas.com/community/threads/help-on-yaml-1-3-executing.115091/",target:"_blank",rel:"noreferrer"},{default:u(()=>[e("Help on yaml:1:3: executing ?? | TrueNAS Community")]),_:1}),e(" 这是因为安装的应用正在升级到的最新版本的、所依赖的 TrueCharts 已经不再支持 Bluefin，只支持 Cobia，需要更新 TrueNAS SCALE 到 Cobia 发行才行。")]),C,a(r),f,a(l)])}const P=c(_,[["render",v]]);export{E as __pageData,P as default};
